src/brasil/gov/agenda/browser/agenda.py

```diff
@@ -34,9 +37,22 @@ def update(self):
         self.workflow = plone_tools.workflow()
         self.editable = context_state.is_editable()
 
+    def results(self, b_size=16):
+        """Retorna as ultimas agendas diárias"""
+        query = {
+            'context': self.context,
+            'object_provides': IAgendaDiaria,
+            'sort_on': 'Date',
+            'sort_order': 'reverse',
+        }
+        b_start = int(self.request.get('b_start', 0))
+        results = api.content.find(**query)
+        results = IContentListing(results)
+        results = Batch(results, b_size, b_start)
+        return results
+
     def __call__(self):
         mapply(self.update, (), self.request)
```

Comment:
what's the function of this `mapply()` thing?

Reply:
no idea

Reply:
This is the code https://github.com/zopefoundation/zope.publisher/blob/master/src/zope/publisher/publish.py#L67-L121
I still don't have an idea after reading it why we need this..

Reply:
and this is the commit when it was  inserted https://github.com/plonegovbr/brasil.gov.agenda/commit/2d17110a235aa8acf8ae9516e5843c0b4feae19b

Reply:
I think we can remove it now that we do a redirect to the **AgendaDiaria** object